network/hostnetwork/transport.go

```diff
@@ -0,0 +1,256 @@
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package hostnetwork
+
+import (
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	"github.com/insolar/insolar/configuration"
+	"github.com/insolar/insolar/core"
+	"github.com/insolar/insolar/log"
+	"github.com/insolar/insolar/network"
+	"github.com/insolar/insolar/network/transport"
+	"github.com/insolar/insolar/network/transport/host"
+	"github.com/insolar/insolar/network/transport/packet"
+	"github.com/insolar/insolar/network/transport/packet/types"
+	"github.com/insolar/insolar/network/transport/relay"
+	"github.com/pkg/errors"
+)
+
+type hostTransport struct {
+	started   uint32
+	transport transport.Transport
+	origin    *host.Host
+	handlers  map[types.PacketType]network.RequestHandler
+}
+
+type builder struct {
+	sender *host.Host
+	t      types.PacketType
+	data   interface{}
+}
+
+func (b *builder) Type(packetType types.PacketType) network.RequestBuilder {
+	b.t = packetType
+	return b
+}
+
+func (b *builder) Data(data interface{}) network.RequestBuilder {
+	b.data = data
+	return b
+}
+
+func (b *builder) GetSender() core.RecordRef {
+	return b.sender.NodeID
+}
+
+func (b *builder) GetSenderHost() *host.Host {
+	return b.sender
+}
+
+func (b *builder) GetType() types.PacketType {
+	return b.t
+}
+
+func (b *builder) GetData() interface{} {
+	return b.data
+}
+
+func (b *builder) Build() network.Request {
+	return b
+}
+
+type packetWrapper packet.Packet
+
+func (p *packetWrapper) GetSender() core.RecordRef {
+	return p.Sender.NodeID
+}
+
+func (p *packetWrapper) GetSenderHost() *host.Host {
+	return p.Sender
+}
+
+func (p *packetWrapper) GetType() types.PacketType {
+	return p.Type
+}
+
+func (p *packetWrapper) GetData() interface{} {
+	return p.Data
+}
+
+type future struct {
+	transport.Future
+}
+
+func (f future) Response() <-chan network.Response {
+	in := transport.Future(f).Result()
+	out := make(chan network.Response, cap(in))
+	go func() {
+		for packet := range in {
+			out <- (*packetWrapper)(packet)
+		}
+	}()
+	return out
+}
+
+func (f future) GetResponse(duration time.Duration) (network.Response, error) {
+	select {
+	case result, ok := <-f.Result():
+		if !ok {
+			return nil, transport.ErrChannelClosed
+		}
+		return (*packetWrapper)(result), nil
+	case <-time.After(duration):
+		f.Cancel()
+		return nil, transport.ErrTimeout
+	}
+}
+
+func (f future) GetRequest() network.Request {
+	request := transport.Future(f).Request()
+	return (*packetWrapper)(request)
+}
+
+// Listen start listening to network requests, should be started in goroutine.
+func (h *hostTransport) Listen() error {
```

Comment:
`.Listen()` and `.Disconnect()` methods are very similar to Component's `.Start()` and `.Stop()` methods.
Could we transform transport to Component?